### PR TITLE
building intermediary metadata snippets for each script

### PR DESCRIPTION
**Pull request by martinraison**
@2is10 as previously discussed, I'm using temporary files instead of global data structures.

### CODE REVIEW

File: packrat/gulp/vinyl-map.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  function flush(cb) {
+    check(done = cb)
+  }
+
+  function check() {
+    if (!pending && done) {
+      done();
+    }
+  }
+}
```
Comment (martinraison) [inline packrat/gulp/vinyl-map.js:64]: We can switch back to the default vinyl-map if https://github.com/hughsk/vinyl-map/pull/6 gets accepted

**Comment by 2is10** [comment]
Cool. Does this make meta build faster after a single .js file edit?

Nice work fixing vinyl-map.

**Comment by martinraison** [comment]
I haven't benchmarked it since the meta build is really not the bottleneck. Currently a single .js file edit triggers a complete rebuild of all the scripts (contrary to what happens after a .css file edit). However, I just discovered the `gulp-cached` module, which may be the solution to many of our problems (I don't want to speak too soon, though). Pull request coming :)

**Comment by 2is10** [comment]
Okay, thanks. Just wanted to make sure we weren’t doing any premature optimizations. I’m pretty confident that we’ll eventually determine that this change has significant, measurable benefit.

**Pull request merged**